src/api.js

```diff
@@ -18,10 +19,66 @@ const getFolderContents = async (siteName, folderName, subfolderName) => {
     return await axios.get(`${BACKEND_URL}/sites/${siteName}/folders?path=_${folderName}${subfolderName ? `/${subfolderName}` : ''}`);
 }
 
-// EditNavBar
+// EditPage
+const getPageApiEndpoint = (isResourcePage, isCollectionPage, collectionName, subfolderName, fileName, siteName, resourceName) => {
+    if (isCollectionPage) {
+        return `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections/${collectionName}/pages/${encodeURIComponent(`${subfolderName ? `${subfolderName}/` : ''}${fileName}`)}`
+    }
+    if (isResourcePage) {
+        return `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources/${resourceName}/pages/${fileName}`
+    }
+    return `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages/${fileName}`
+}
+
+const getEditPageData = async (isResourcePage, isCollectionPage, collectionName, subfolderName, fileName, siteName, resourceName) => {
```

Comment:
As discussed, we should split up the `getEditPageData` API function, since retrieval of the CSP and directory config file aren't directly related to retrieving the `EditPage` content. 
- CSP retrieval should be its own API function
- we don't need a separate API function for directory config retrieval since we already have `getDirectoryFile` 

Once that is down, we'll set up new `useQuery` functions for CSP retrieval and directory config file retrieval. This allows us to make use of `react-query's` caching functionality. Caching is important since the CSP is reusable across all pages in the repo, and directory config is retrieved for the `Folders` / sub-folders layout as well as for all collection pages. Therefore, we should be careful to specify the same query keys for the directory config `useQuery`.

Reply:
Good point, resolved in https://github.com/isomerpages/isomercms-frontend/pull/379/commits/90a952df4609cc81d713e8eb5e494d9046c160d5